### PR TITLE
refactor: update test coverage

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
Separates the segment service config into its own service module, and added test coverage.
Padded up a bit more unit testing in the data model in general.
Fixed one of the segment triggers that wasn't working.

** [Edit] update ** I finished the rest of the unit tests as well

image

JIRA: 
* [AU-33](https://openedx.atlassian.net/browse/AU-33)
* [AU-34](https://openedx.atlassian.net/browse/AU-34)

**Developer Checklist**
- [ ] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Review by nsprenkle — APPROVED** [review]
:shipit: 💯
